### PR TITLE
Testing

### LINKED ISSUE #185: Latest upgrade overwrite configuration file application.ini
Hello, 
3 days ago, I upgraded from `2.37.3~ynh1` to latest version `2.38.2~ynh1`, and the configuration was the lost. 
I found that the file `/var/www/snappymail/app/data/_data_/_default_/configs/application.ini` is overwritten during the upgrade. 
I saw that the line about this config file is commented in the upgrade script: https://github.com/YunoHost-Apps/snappymail_ynh/blob/82f8fb425c66ba12566d1070c6a2d22d911fcfe8/scripts/upgrade#L33 I am not sure why. 

I think we should at least keep the old file and not the one provided within the sources, because this last one has only default values. 

Looking at upgrade logs I see that we specify to keep this file, but maybe it's a bug :thinking: 
```
2024-10-18 17:32:52,338: DEBUG - + ynh_setup_source --dest_dir=/var/www/snappymail/app --full_replace --keep=app/data/_data_/_default_/configs/application.ini
2024-10-18 17:32:52,401: DEBUG - + keep=app/data/_data_/_default_/configs/application.ini
```
full logs: https://paste.yunohost.org/temolowani

Thanks for your advices, I might send a PR after a few chat :slightly_smiling_face:

### LINKED ISSUE #187: SSO is broken
### Describe the bug

When logged in with YunoHost, you shouldn't be asked to log in SnappyMail.

### Context

- YunoHost version: 12.0.6

### Steps to reproduce

- Login to YunoHost portal
- Click on SnappyMail

### Expected behavior

https://github.com/YunoHost-Apps/snappymail_ynh/blob/master/conf/sso.php should log you in snappymail using YunoHost HTTP headers or something
This might be related to YunoHost 12 ssowat changes.

### PR DESCRIPTION
**Pull request by Tagadda**
## Problem

- *Description of why you made this PR*

## Solution

- Fix #187 
- Fix #185 
- Removes YunoHost tile

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by Tagadda** [comment]
!bump

**Comment by Tagadda** [comment]
!testme

File: scripts/upgrade [hunk 1]
```diff
@@ -22,7 +22,7 @@ fi
 #=================================================
 ynh_script_progression "Upgrading source files..."
 
-ynh_setup_source --dest_dir="$install_dir/app" --full_replace --keep="app/data/_data_/_default_/configs/application.ini"
```
Comment (ericgaspar) [inline scripts/upgrade:25]: why are we removing `/app` path ?
if the path is `$install_dir/app/data/_data_/_default_/configs/application.ini` we should keep `/app`, no?

Comment (Tagadda) [reply scripts/upgrade:25]: According to the commit message, `keep` path is relative to `dest_dir`, not `install_path`

**Pull request merged**